consensus/votepower/roster.go

```diff
@@ -144,78 +159,66 @@ func AggregateRosters(rosters []RosterPerShard) map[common.Address]Staker {
 	return result
 }
 
-// JSON dump
-func (r *Roster) JSON() string {
-	v := map[string]stakedVoter{}
-	for k, value := range r.Voters {
-		v[k.Hex()] = value
-	}
-	c := struct {
-		Voters map[string]stakedVoter `json:"voters"`
-		Our    string                 `json:"ours"`
-		Their  string                 `json:"theirs"`
-		Raw    string                 `json:"raw-total"`
-	}{
-		v,
-		r.OurVotingPowerTotalPercentage.String(),
-		r.TheirVotingPowerTotalPercentage.String(),
-		r.RawStakedTotal.String(),
-	}
-	b, _ := json.Marshal(&c)
-	return string(b)
-}
-
 // Compute creates a new roster based off the shard.SlotList
-func Compute(staked shard.SlotList) (*Roster, error) {
-	roster := NewRoster()
+func Compute(subComm *shard.Committee) (*Roster, error) {
+	roster, staked := NewRoster(subComm.ShardID), subComm.Slots
+
 	for i := range staked {
-		if staked[i].EffectiveStake == nil {
-			roster.HmySlotCount++
-		} else {
-			roster.RawStakedTotal = roster.RawStakedTotal.Add(
-				*staked[i].EffectiveStake,
+		if e := staked[i].EffectiveStake; e != nil {
+			roster.RawStakedTotal.Add(
+				roster.RawStakedTotal, e.TruncateInt(),
```

Comment:
RawStakedTotal is a bad name. It's the TotalEffectiveStake, and it's composed of Decs. We should keep using dec rather than int.

Reply:
But the stake itself is token and token can only be integer, right? So decimal only needed in actual computations when floats arrive, like the epos computation itself, but summing up what people actually stake is only integer token, no?